Changes made

greykoalacode opened the pull request:
## Description of the change

> Added the Submit tab of Contests page, where problem details and respective language details are given with either the solution file/ code as text. This is being stored and on validation, loading screen is shown, else an error alert is displayed. 
> Alert component made in case , we need alerts to be displayed, it is reusable as we can give the message and type of message as props. 

## Type Of Change
- [ ] Bug Fix (non-breaking change that fixes an issue)
- [*] New Feature (non-breaking change that adds functionality)
- [ ] Breaking Change (fix or feature that would cause existing functionality to not work as expected)

## Related Issues

> Fix [#<ISSUE_NUMBER>](<ISSUE_LINK>) 

## Development & Code Review 
- [*] Fixed all the Eslint errors (run `npm run lint-fix`)
- [*] All ESLint rules passes locally (run `npm run lint`)
- [*] The code changed/added has been covered with tests
- [ ] All the test cases added/changed are passing

greykoalacode commented:
Hey , i  have done the following:
- [*] Merge your branch with the master so that you get updated code.
- [*] We already have an alert component called MessageCard inside common folder. You can directly use that, no need to write a new one.
- [*] We also have a Spinner component already inside common folder.
- [ ] Don't apply box-shadow to the container. It looks weird in my opinion.
- [*] Use material.io grid to align the select boxes and files if you want.
- [*] You have added a huge css file for SubmitFile. But you have achieve the same with few tachyons classes. There is no specific need to add a separate css file. You just have to add margins and paddings. The problem with a custom css is that you have to write media queries and other adjustments. But tachyons provides atomic classes for the same so you can directly use them.
- [*] Make the select boxes of Select problem and Select Language side by side. It will look better that way. You can refer to createProblem code inside admin panel to get an idea about the usage of Grid and aligning items with Grid.

ac030540 commented on src/Components/drawer/contests/submit/SubmitFile.jsx:
```diff
@@ -0,0 +1,191 @@
+export default function SubmitFile() {
+// initial State declaration
+  const [value, setValue] = useState('default');
+  const [uploadMethod, setMethod] = useState('file');
```
Make the function to update the state as setUploadMethod. So that it can be easily co-related to the state it changes. Do same thing with other state update functions.

ac030540 commented on src/Components/drawer/contests/submit/SubmitFile.jsx:
```diff
@@ -0,0 +1,191 @@
+  // function to render loading page / the submit page
+  if (isUploading) {
+    return (
+      <div className="container">
```
Remove this container class. It doesn't look good IMO. Also remove the CSS file as you probably won't need it then.

ac030540 commented on src/Components/drawer/contests/submit/SubmitFile.jsx:
```diff
@@ -0,0 +1,191 @@
+      {/* Alert is a mini component used here ,
+      we pass message and type of alert
+      and it sets it accordingly */}
+      {hasError
```
Don't use MessageCard this way. Many things are handled by the component internally. Refer to the places where I have used it. Pass all the props which are required by the component and you probably won't even need this hasError state.

ac030540 commented on package.json:
```diff
@@ -7,11 +7,13 @@
     "@material-ui/core": "^4.8.3",
     "@material-ui/pickers": "^3.2.8",
     "@material/elevation": "^0.44.0",
+    "@material/linear-progress": "^5.1.0",
```
Remove unused packages from package.json

greykoalacode commented:
Hey , i have made the following changes:
- Added desktopColumns, tabletColumns and phoneColumns in Cell to make them more reponsive and as the look you had wanted..
- Removed unwanted tachyons in SubmitFile.jsx and removed SubmitFile.css
- Read about MessageCard and assigned it in a better way, removed hasError parts , also the console error got off. 
- Edited the names of setting State functions, wherever seemed necessary
- paddings and margins are mostly standardised, except few places where a single component needed little different.

ac030540 commented on package.json:
```diff
@@ -4,14 +4,12 @@
   "private": true,
   "dependencies": {
     "@date-io/date-fns": "^1.3.13",
-    "@material-ui/core": "^4.8.3",
-    "@material-ui/pickers": "^3.2.8",
-    "@material/elevation": "^0.44.0",
```
Three of these packages are needed. That's the same reason why travis CI build failed.

greykoalacode commented:
Hey I have reinstalled those packages, I didnt read from properly

ac030540 commented:
> Hey I have reinstalled those packages, I didnt read from properly

You build still failed. Figure out what is the error.

ac030540 commented on src/Components/drawer/contests/submit/SubmitFile.jsx:
```diff
@@ -0,0 +1,188 @@
+import Select, { Option } from '@material/react-select';
+import TextField, { Input } from '@material/react-text-field';
+import contestDetails from './contestDetails';
+import '@material/react-menu-surface/dist/menu-surface.css';
+import '@material/react-menu/dist/menu.css';
+import '@material/react-select/dist/select.css';
```
No need to do these imports. These are already imported in `App.scss`.

ac030540 commented on src/Components/drawer/contests/submit/SubmitFile.jsx:
```diff
@@ -0,0 +1,188 @@
+const languages = ['C', 'C++', 'Python 2', 'Python 3', 'Java', 'Go', 'Javascript'];
+
+
+export default function SubmitFile() {
```
Make it an arrow function. Move the export default statement at the end of the file. It's just for code uniformity.

ac030540 replied:
> Make it an arrow function. Move the export default statement at the end of the file. It's just for code uniformity.

You haven't done the above changes.

ac030540 commented on src/Components/drawer/contests/submit/SubmitFile.jsx:
```diff
@@ -0,0 +1,188 @@
+  // function to check validation
+  const validationCheck = () => {
+    if (value !== 'default' && lang !== 'default' && (file.length !== 0 || code.length !== 0)) {
+      onSubmitTrigger(true);
```
Here there is no need to write a separate function to setIsUploading as true as you are using it only once. So here you can directly write `setIsUploading(true)`. Also the function is of one line and the statement to update the state is also of one line so it is not giving any significant reduction in LOC (Lines Of Code).

ac030540 commented on src/Components/drawer/contests/submit/SubmitFile.jsx:
```diff
@@ -0,0 +1,188 @@
+  return (
+    <div className="">
+      <div className="w-100 ml4 mr2 pr5">
+        <MessageCard message={message} messageType={messageType} setMessageType={setMessageType} />
```
![Screenshot 2020-04-08 at 9 58 54 AM](https://user-images.githubusercontent.com/40123795/78744702-db94f000-797f-11ea-8e0f-afb7fcfabd18.png)
Here the MessageCard component is having a different padding/margin on smaller screens. Fix that. One reason can be because you have kept it outside the grid while other components are inside the grid. Remember grid has it's own padding/margin.

ac030540 commented on src/Components/drawer/contests/submit/SubmitFile.jsx:
```diff
@@ -0,0 +1,188 @@
+        <Row>
+          <Cell tabletColumns="2">
+            <div className="">
+              <Select
+                required
+                notchedOutlineClassName="pa2"
+                enhanced
+                outlined
+                label="Problem"
+                className="w-100"
+                value={value}
+                onEnhancedChange={onEnhancedChange}
```
![Screenshot 2020-04-08 at 9 58 54 AM](https://user-images.githubusercontent.com/40123795/78744871-47775880-7980-11ea-9be1-6ff8f17a5055.png)
The text of select boxes is moving out of the box. It is not an issue from your end. Try updating, the react-select, react-menu, react-menu-surface. If the issue is still not fixed then you can use different select box provided by material.io. You can refer the code in `ResetSubmissionTableEntry.jsx` to understand how to use that select.

ac030540 replied:
There is a difference in the padding/margin of the SubmitFile component and the ContestDetails and Announcement component. Fix that.

ac030540 commented on src/Components/drawer/contests/submit/SubmitFile.jsx:
```diff
@@ -0,0 +1,188 @@
+        </Row>
+        <Row>
+          <Cell>
+            <button
+              className="ml2 mdc-button mdc-ripple-upgraded mdc-button--raised"
+              type="submit"
+              onClick={() => { validationCheck(); }}
+            >
```
Rather than adding classes, you can directly use the Button component provided by material.io. You can refer to the code of signin component to understand how to use it.

greykoalacode commented:
Hey Ashok, check the changes have been done. Let me know if anything needed to be changed

ac030540 commented on src/Components/drawer/contests/submit/SubmitFile.jsx:
```diff
@@ -0,0 +1,193 @@
+        <Row>
+          <Cell tabletColumns="4">
+            <div className="">
+              <Select
+                required
+                notchedOutlineClassName="pa2"
+                enhanced
+                outlined
+                label="Problem"
+                className="w-100"
+                value={value}
+                onEnhancedChange={onEnhancedChange}
+              >
+                <Option className="opt" value="default">Choose Problem</Option>
+                {
```
![Screenshot 2020-04-09 at 9 40 20 PM](https://user-images.githubusercontent.com/40123795/78916688-7ccc9500-7aab-11ea-857c-386146f2ef24.png)
The problem with this select is that, when the text size is too large then the text overflows. Try updating the version of select component or use a different type of select provided by material.io.

Swap76 commented:
@greykoalacode Congratulations for the first merged PR

Looking forward to your work on the repo in the future :-)

greykoalacode commented:
Thank you @Swap76  and @ac030540  for giving me opportunities and helping me learn all these ! Will do more in future! :-)
